src/js/utils.js

```diff
@@ -0,0 +1,53 @@
+/*!
+Copyright 2015 OCAD University
+
+Licensed under the New BSD license. You may not use this file except in
+compliance with this License.
+
+You may obtain a copy of the License at
+https://github.com/gpii/universal/LICENSE.txt
+*/
+
+(function ($, fluid) {
+
+    "use strict";
+
+    fluid.registerNamespace("gpii.d3");
+
+    /**
+     * Convert jQuery DOM elements to D3 DOM elements.
+     * @param domElem - a jQuery DOM element or an array of jQuery DOM elements
+     */
+    gpii.d3.jQueryToD3 = function (elem) {
+        return d3.selectAll(elem.toArray());
+    };
+
+    gpii.d3.addD3Listeners = function (jQueryElem, eventName, listener, that) {
+        var d3Elem = gpii.d3.jQueryToD3(jQueryElem);
```

Comment:
What if an integrator swapped the jQueryToD3 invoker in the gpii.d3ViewComponent to a different function. They would still get the old one when they call the addD3Listeners invoker.